app/models/manager_refresh/save_collection/helper.rb

```diff
@@ -34,8 +34,9 @@ def save_inventory_collection!(inventory_collection, association)
         inventory_objects_index[index] = inventory_object
       end
 
-      unique_index_keys = inventory_collection.manager_ref_to_cols
-      unique_db_indexes = Set.new
+      unique_index_keys      = inventory_collection.manager_ref_to_cols
+      unique_db_indexes      = Set.new
+      unique_db_primary_keys = Set.new
```

Comment:
actually no - it's just rails that decided to have `id` as a primary key. But that was not my point.

In this case you are using `unique_db_primary_keys` as a lookup / hash table and `unique_index_keys` is an array of keys.